yt/data_objects/derived_quantities.py

```diff
@@ -623,7 +630,7 @@ def reduce_intermediate(self, values):
         # The values get turned into arrays here.
         return [
             self.data_source.ds.arr([mis.min(), mas.max()])
-            for mis, mas in zip(values[::2], values[1::2])
+            for mis, mas in zip(values[::2], values[1::2], strict=True)
```

Comment:
This is only safe if `len(values)%2==0` is guaranteed. If not, `strict=False` matches the existing behavior but is possibly buggy (one value is silently dropped)

Reply:
I think using strict here is the right move, as a check on argument inputs.